### PR TITLE
Adding in make_primary method

### PR DESCRIPTION
**Pull request by roborourke**
This mapping method updates the primary domain for a site eg. it's domain value in the wp_blogs table and creates an alias from the previous value, the value in wp_blogs can then be assumed to be the primary domain.

Ok to review this @rmccue?

### CODE REVIEW

File: class-mapping.php [hunk 1]
```diff
@@ -102,6 +102,38 @@ public function set_active( $active ) {
...
+			// These are just a visual update for the site admin
+			$url = esc_url( $this->get_domain() );
+			update_blog_option( $site->blog_id, 'home', $url );
+			update_blog_option( $site->blog_id, 'siteurl', $url );
+
+			// Remove current mapping
+			$this->delete();
+		} else {
+			return $mapping;
+		}
```
Comment (rmccue) [inline class-mapping.php]: This conditional should be flipped and the error returned early instead.

File: class-mapping.php [hunk 2]
```diff
@@ -102,6 +102,38 @@ public function set_active( $active ) {
 	}
 
 	/**
+	 * Make this mapping the primary domain, eg. set the homeurl for the site
+	 *
+	 * @return bool|\WP_Error True if we created the old mapping, false if we didn't need to, or WP_Error if an error occurred
+	 */
+	public function make_primary() {
```
Comment (rmccue) [inline class-mapping.php]: Wondering if this should maybe a function instead of a method, given that it destroys the object.

**Comment by roborourke** [comment]
Hmm, although it's an explicit method name have a verb has its advantages in terms of naming hooks and a neat REST endpoint for it. Is there a word other than `promote` that would indicate that something else is being `demoted` at the same time?

**Comment by mrjarbenne** [comment]
> Is there a word other than promote that would indicate that something else is being demoted at the same time?

- supersede
- supplant

**Comment by roborourke** [comment]
Hmm. Think we're going down a rabbit hole here. 'Make Primary' is the most explicit and true to the data manipulation going on under the hood. I'll add the action like the other methods have and merge this in. We can always change it later.

**Comment by rmccue** [comment]
@roborourke Forgot to reply to you on the naming point; "make primary" is perfect and clear, anything else is going to be a bit too obtuse I think.

**Comment by roborourke** [comment]
No worries! Yeah great minds. Will ping you about the rest API changes on
Monday

**Pull request merged**